### PR TITLE
surface code locations in the frontend

### PR DESCRIPTION
**Pull request by breadchris**
screenshot-localhost_4455-2022 12 15-21_41_01

Code locations are returned to the frontend and displayed in the package card. Code locations are grouped together based off of finding the closest parent directory with a manifest.

### CODE REVIEW

**Comment by breadchris** [comment]
test

**Comment by breadchris** [comment]
test

**Comment by ajvpot** [comment]
```
graphql-engine_1  | {"level":"error","msg":"skipping applying migrations on database lunatrace, encountered: \n{\n  \"code\": \"postgres-error\",\n  \"error\": \"query execution failed\",\n  \"internal\": {\n    \"arguments\": [],\n    \"error\": {\n      \"description\": null,\n      \"exec_status\": \"FatalError\",\n      \"hint\": null,\n      \"message\": \"relation \\\"vulnerability_cwe_vulnerability_id_cwe_id_key\\\" already exists\",\n      \"status_code\": \"42P07\"\n    },\n    \"prepared\": false,\n    \"statement\": \"ALTER TABLE \\\"vulnerability\\\".\\\"vulnerability_cwe\\\" ADD CONSTRAINT \\\"vulnerability_cwe_vulnerability_id_cwe_id_key\\\" UNIQUE (\\\"vulnerability_id\\\", \\\"cwe_id\\\");\\n\"\n  },\n  \"path\": \"$\"\n}","time":"2022-12-16T05:49:25Z"}
```

**Comment by ajvpot** [comment]
> "relation \\\"vulnerability_cwe_vulnerability_id_cwe_id_key\\\" already exists

File: lunatrace/bsl/backend/src/hasura-api/actions/create-new-build.ts [hunk 1]
```diff
@@ -59,7 +60,7 @@ export async function createNewBuild(repoGithubId: number, buildInfo: SnapshotBu
           source_type: buildInfo.sourceType,
           git_hash: buildInfo.gitCommit,
           git_branch: buildInfo.gitBranch,
-          git_remote: buildInfo.cloneUrl,
+          git_remote: removeAuthFromGitUrl(buildInfo.cloneUrl),
```
Comment (ajvpot) [inline lunatrace/bsl/backend/src/hasura-api/actions/create-new-build.ts:63]: :sunglasses:

File: lunatrace/bsl/backend/src/hasura-api/graphql/get-tree-from-build.graphql [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
-                child_id
-                analysis_results(where: $analysis_results_where, order_by: {finding_source_version: desc}) {
-                  finding_source_version
-                  finding_source
-                  finding_type
+                child_edges {
+                    ...ManifestDependencyEdge
                 }
             }
+            child_edges_recursive {
```
Comment (ajvpot) [inline lunatrace/bsl/backend/src/hasura-api/graphql/get-tree-from-build.graphql:32]: do we need to fetch this twice?

**Review by ajvpot — APPROVED** [review]

File: lunatrace/bsl/backend/src/hasura-api/graphql/get-tree-from-build.graphql (hunk 2 above)
Comment (breadchris) [reply lunatrace/bsl/backend/src/hasura-api/graphql/get-tree-from-build.graphql:29]: you are right, this was from debugging. removed.

**Pull request merged**